### PR TITLE
feat: add JwtQueryKey to read token as query parameter

### PR DESCRIPTION
**Pull request by simonpahl**
This PR makes it possible to extract the JWT from a query parameter specified in the JwtQueryKey configuration.

### CODE REVIEW

**Review by eshepelyuk — CHANGES_REQUESTED** [review]
Please add tests.

**Comment by simonpahl** [comment]
Tests were added.

File: jwt.go [hunk 1]
```diff
@@ -484,6 +490,11 @@ func (jwtPlugin *JwtPlugin) extractTokenFromCookie(request *http.Request) (strin
 	return cookie.Value, nil
 }
 
+func (jwtPlugin *JwtPlugin) extractTokenFromQuery(request *http.Request) (string, error) {
+	parameter := request.URL.Query().Get(jwtPlugin.jwtQueryKey)
```
Comment (eshepelyuk) [inline jwt.go:494]: What happens if parameter not found in query ? Will it result in an error ?
This function should return proper error if query parameter is missing, as extract from header and extract from cookie do.

File: jwt_test.go [hunk 2]
```diff
@@ -983,3 +983,61 @@ func TestTokenFromCookieNotConfigured(t *testing.T) {
...
+	req.URL.RawQuery = query.Encode()
+
+	jwt.ServeHTTP(recorder, req)
+
+	if nextCalled == false {
+		t.Fatal("next.ServeHTTP was not called")
+	}
+}
+
+func TestTokenFromQueryNotConfigured(t *testing.T) {
```
Comment (eshepelyuk) [inline jwt_test.go:1016]: This test should enable extract from query parameter but not provide a query parameter in request. An then analyze that http status is 403 and response body says that query param is missing.

**Review by eshepelyuk — CHANGES_REQUESTED** [review]
Left some comments inline.

File: jwt.go (hunk 1 above)
Comment (simonpahl) [reply jwt.go]: Done.

File: jwt_test.go (hunk 2 above)
Comment (simonpahl) [reply jwt_test.go:1052]: I added the test also for cookies.

**Review by eshepelyuk — DISMISSED** [review]
Plz rebase and squash this to single commit, otherwise LGTM.

**Comment by eshepelyuk** [comment]
Thanx for the contribution @simonpahl

**Pull request merged**